src/Microsoft.OpenApi/Validations/Rules/OpenApiSchemaRules.cs

```diff
@@ -55,6 +57,72 @@ public static class OpenApiSchemaRules
                     context.Exit();
                 });
 
+        /// <summary>
+        /// Validates OneOf Discriminator
+        /// </summary>
+        public static ValidationRule<OpenApiSchema> ValidateOneOfDiscriminator =>
+            new ValidationRule<OpenApiSchema>(
+                (context, schema) =>
+                {
+                    // oneOf
+                    context.Enter("oneOf");
+
+                    if (schema.OneOf != null && schema.Discriminator != null)
```

Comment:
The spec also says this:

"When using the discriminator, inline schemas will not be considered."

To be honest, this sentence is a bit ambiguous. It doesn't make sense to have an inline schema but the spec doesn't disallow it per se. 

For our purpose, I'd suggest we perform this check ONLY when the reference field is also present.

ie. Change the if condition to the version below that's a bit stricter:

`if (schema.OneOf != null && schema.Discriminator != null && schema.Reference != null)`